gameobjects/units/cUnitUtils.cpp

```diff
@@ -22,13 +22,13 @@ int cUnitUtils::findUnit(int type, int iPlayerId) {
 // find the first unit of type belonging to player Id, and is not the same Id as iIgnoreUnitId.
 int cUnitUtils::findUnit(int type, int iPlayerId, int iIgnoreUnitId) {
 	for (int i=0; i < MAX_UNITS; i++) {
-		if (i != iIgnoreUnitId) {
-			if (unit[i].isValid()) {
-				if (unit[i].iPlayer == iPlayerId) {
-					return i;
-				}
-			}
-		}
+    if (i == iIgnoreUnitId || !unit[i].isValid()) {
+      continue;
+    }
+
+    if (unit[i].iPlayer == iPlayerId && unit[i].iType == type) {
```

Comment:
Here, I decided to use the type parameter. 
This function isn't used (neither is the class), but I didn't want to throw away work in case this is the direction we want to take with the split-up of the large cUnit.cpp.